File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/AgentFlowDiagram.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Diagrammatic representation of the Agent Flow with all node and edge details
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AgentFlowDiagram.Builder.class)

public final class AgentFlowDiagram  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "displayName", "description", "nodes", "edges", "toolMap", "guardrailsMap"})
    public AgentFlowDiagram(String key, String displayName, String description, java.util.Map<String, AgentFlowNode> nodes, java.util.Map<String, AgentFlowEdge> edges, java.util.Map<String, Tool> toolMap, java.util.Map<String, GuardrailsConfiguration> guardrailsMap) {
        super();
        this.key = key;
        this.displayName = displayName;
        this.description = description;
        this.nodes = nodes;
        this.edges = edges;
        this.toolMap = toolMap;
        this.guardrailsMap = guardrailsMap;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The unique identifier (UUID) of the Agent flow
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * The unique identifier (UUID) of the Agent flow
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * AgentFlow name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

        /**
         * AgentFlow name.
         * @param displayName the value to set
         * @return this builder
         **/
        

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
            /**
     * AgentFlow description.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * AgentFlow description.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * Mapping of nodeId to node objects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("nodes")
private java.util.Map<String, AgentFlowNode> nodes;

        /**
         * Mapping of nodeId to node objects.
         * @param nodes the value to set
         * @return this builder
         **/
        

public Builder nodes(java.util.Map<String, AgentFlowNode> nodes) {
    this.nodes = nodes;
    return this;
}
            /**
     * Mapping of edgeId to edge objects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("edges")
private java.util.Map<String, AgentFlowEdge> edges;

        /**
         * Mapping of edgeId to edge objects.
         * @param edges the value to set
         * @return this builder
         **/
        

public Builder edges(java.util.Map<String, AgentFlowEdge> edges) {
    this.edges = edges;
    return this;
}
            /**
     * A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("toolMap")
private java.util.Map<String, Tool> toolMap;

        /**
         * A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.
         * @param toolMap the value to set
         * @return this builder
         **/
        

public Builder toolMap(java.util.Map<String, Tool> toolMap) {
    this.toolMap = toolMap;
    return this;
}
            /**
     * A hash map with key=guardrails config name, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("guardrailsMap")
private java.util.Map<String, GuardrailsConfiguration> guardrailsMap;

        /**
         * A hash map with key=guardrails config name, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.
         * @param guardrailsMap the value to set
         * @return this builder
         **/
        

public Builder guardrailsMap(java.util.Map<String, GuardrailsConfiguration> guardrailsMap) {
    this.guardrailsMap = guardrailsMap;
    return this;
}


        public AgentFlowDiagram build() {
            AgentFlowDiagram model = new AgentFlowDiagram(this.key
                , this.displayName
                , this.description
                , this.nodes
                , this.edges
                , this.toolMap
                , this.guardrailsMap);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AgentFlowDiagram model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.nodes(model.getNodes());
    this.edges(model.getEdges());
    this.toolMap(model.getToolMap());
    this.guardrailsMap(model.getGuardrailsMap());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The unique identifier (UUID) of the Agent flow
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * The unique identifier (UUID) of the Agent flow
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * AgentFlow name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * AgentFlow name.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * AgentFlow description.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * AgentFlow description.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * Mapping of nodeId to node objects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("nodes")
    private final java.util.Map<String, AgentFlowNode> nodes;

        /**
     * Mapping of nodeId to node objects.
     * @return the value
     **/
    
    public java.util.Map<String, AgentFlowNode> getNodes() {
        return nodes;
    }


        /**
     * Mapping of edgeId to edge objects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("edges")
    private final java.util.Map<String, AgentFlowEdge> edges;

        /**
     * Mapping of edgeId to edge objects.
     * @return the value
     **/
    
    public java.util.Map<String, AgentFlowEdge> getEdges() {
        return edges;
    }


        /**
     * A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("toolMap")
    private final java.util.Map<String, Tool> toolMap;

        /**
     * A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.
     * @return the value
     **/
    
    public java.util.Map<String, Tool> getToolMap() {
        return toolMap;
    }


        /**
     * A hash map with key=guardrails config name, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("guardrailsMap")
    private final java.util.Map<String, GuardrailsConfiguration> guardrailsMap;

        /**
     * A hash map with key=guardrails config name, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.
     * @return the value
     **/
    
    public java.util.Map<String, GuardrailsConfiguration> getGuardrailsMap() {
        return guardrailsMap;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AgentFlowDiagram(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", nodes=").append(String.valueOf(this.nodes));
        sb.append(", edges=").append(String.valueOf(this.edges));
        sb.append(", toolMap=").append(String.valueOf(this.toolMap));
        sb.append(", guardrailsMap=").append(String.valueOf(this.guardrailsMap));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AgentFlowDiagram)) {
            return false;
        }

        AgentFlowDiagram other = (AgentFlowDiagram) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.nodes, other.nodes) &&
            java.util.Objects.equals(this.edges, other.edges) &&
            java.util.Objects.equals(this.toolMap, other.toolMap) &&
            java.util.Objects.equals(this.guardrailsMap, other.guardrailsMap);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.nodes == null ? 43 : this.nodes.hashCode());
        result = (result * PRIME) + (this.edges == null ? 43 : this.edges.hashCode());
        result = (result * PRIME) + (this.toolMap == null ? 43 : this.toolMap.hashCode());
        result = (result * PRIME) + (this.guardrailsMap == null ? 43 : this.guardrailsMap.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/agent_flow_diagram.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AgentFlowDiagram(object):
    """
    Diagrammatic representation of the Agent Flow with all node and edge details
    """

    def __init__(self, **kwargs):
        """
        Initializes a new AgentFlowDiagram object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this AgentFlowDiagram.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this AgentFlowDiagram.
        :type display_name: str

        :param description:
            The value to assign to the description property of this AgentFlowDiagram.
        :type description: str

        :param model_version:
            The value to assign to the model_version property of this AgentFlowDiagram.
        :type model_version: str

        :param nodes:
            The value to assign to the nodes property of this AgentFlowDiagram.
        :type nodes: dict(str, AgentFlowNode)

        :param edges:
            The value to assign to the edges property of this AgentFlowDiagram.
        :type edges: dict(str, AgentFlowEdge)

        :param tools_map:
            The value to assign to the tools_map property of this AgentFlowDiagram.
        :type tools_map: dict(str, Tool)

        :param guardrails_map:
            The value to assign to the guardrails_map property of this AgentFlowDiagram.
        :type guardrails_map: dict(str, GuardrailsConfiguration)

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'description': 'str',
            'model_version': 'str',
            'nodes': 'dict(str, AgentFlowNode)',
            'edges': 'dict(str, AgentFlowEdge)',
            'tools_map': 'dict(str, Tool)',
            'guardrails_map': 'dict(str, GuardrailsConfiguration)'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'description': 'description',
            'model_version': 'modelVersion',
            'nodes': 'nodes',
            'edges': 'edges',
            'tools_map': 'toolsMap',
            'guardrails_map': 'guardrailsMap'
        }

        self._key = None
        self._display_name = None
        self._description = None
        self._model_version = None
        self._nodes = None
        self._edges = None
        self._tools_map = None
        self._guardrails_map = None

    @property
    def key(self):
        """
        Gets the key of this AgentFlowDiagram.
        The unique identifier (UUID) of the Agent flow


        :return: The key of this AgentFlowDiagram.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this AgentFlowDiagram.
        The unique identifier (UUID) of the Agent flow


        :param key: The key of this AgentFlowDiagram.
        :type: str
        """
        self._key = key

    @property
    def display_name(self):
        """
        Gets the display_name of this AgentFlowDiagram.
        AgentFlow name.


        :return: The display_name of this AgentFlowDiagram.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this AgentFlowDiagram.
        AgentFlow name.


        :param display_name: The display_name of this AgentFlowDiagram.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        Gets the description of this AgentFlowDiagram.
        AgentFlow description.


        :return: The description of this AgentFlowDiagram.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this AgentFlowDiagram.
        AgentFlow description.


        :param description: The description of this AgentFlowDiagram.
        :type: str
        """
        self._description = description

    @property
    def model_version(self):
        """
        Gets the model_version of this AgentFlowDiagram.
        The model and upgrade compatibility version for this agent flow diagram.


        :return: The model_version of this AgentFlowDiagram.
        :rtype: str
        """
        return self._model_version

    @model_version.setter
    def model_version(self, model_version):
        """
        Sets the model_version of this AgentFlowDiagram.
        The model and upgrade compatibility version for this agent flow diagram.


        :param model_version: The model_version of this AgentFlowDiagram.
        :type: str
        """
        self._model_version = model_version

    @property
    def nodes(self):
        """
        Gets the nodes of this AgentFlowDiagram.
        Mapping of nodeId to node objects.


        :return: The nodes of this AgentFlowDiagram.
        :rtype: dict(str, AgentFlowNode)
        """
        return self._nodes

    @nodes.setter
    def nodes(self, nodes):
        """
        Sets the nodes of this AgentFlowDiagram.
        Mapping of nodeId to node objects.


        :param nodes: The nodes of this AgentFlowDiagram.
        :type: dict(str, AgentFlowNode)
        """
        self._nodes = nodes

    @property
    def edges(self):
        """
        Gets the edges of this AgentFlowDiagram.
        Mapping of edgeId to edge objects.


        :return: The edges of this AgentFlowDiagram.
        :rtype: dict(str, AgentFlowEdge)
        """
        return self._edges

    @edges.setter
    def edges(self, edges):
        """
        Sets the edges of this AgentFlowDiagram.
        Mapping of edgeId to edge objects.


        :param edges: The edges of this AgentFlowDiagram.
        :type: dict(str, AgentFlowEdge)
        """
        self._edges = edges

    @property
    def tools_map(self):
        """
        Gets the tools_map of this AgentFlowDiagram.
        A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.


        :return: The tools_map of this AgentFlowDiagram.
        :rtype: dict(str, Tool)
        """
        return self._tools_map

    @tools_map.setter
    def tools_map(self, tools_map):
        """
        Sets the tools_map of this AgentFlowDiagram.
        A hash map with key=tool key, value=tool definition.   It is used to find the tool definition for a tool reference, where the tool is marked as a reference.


        :param tools_map: The tools_map of this AgentFlowDiagram.
        :type: dict(str, Tool)
        """
        self._tools_map = tools_map

    @property
    def guardrails_map(self):
        """
        Gets the guardrails_map of this AgentFlowDiagram.
        A hash map with key=guardrails config key, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.


        :return: The guardrails_map of this AgentFlowDiagram.
        :rtype: dict(str, GuardrailsConfiguration)
        """
        return self._guardrails_map

    @guardrails_map.setter
    def guardrails_map(self, guardrails_map):
        """
        Sets the guardrails_map of this AgentFlowDiagram.
        A hash map with key=guardrails config key, value=guardrails definition.   It is used to find the guardrails definition for a guardrails reference by name.


        :param guardrails_map: The guardrails_map of this AgentFlowDiagram.
        :type: dict(str, GuardrailsConfiguration)
        """
        self._guardrails_map = guardrails_map

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
